src/ng/compile.js

```diff
@@ -1626,6 +1626,37 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     }
 
     /**
+     * A function generator that is used to support both eager and lazy compilation
+     * linking function.
+     * @param eager
+     * @param $compileNodes
+     * @param transcludeFn
+     * @param maxPriority
+     * @param ignoreDirective
+     * @param previousCompileContext
+     * @returns {Function}
+     */
+    function compilationGenerator(eager, $compileNodes, transcludeFn, maxPriority, ignoreDirective, previousCompileContext) {
```

Comment:
[Not a blocker to land this]
I am somehow torn on having all the parameters here or if we should use `arguments`. i.e.

```js
function compilationGenerator(eager) {
  var compileArguments = Array.slice(arguments);
  compileArguments.shift();
  [...]
    return compile.apply(null, compileArguments);
  [...]
}
```

It is somehow more code, but it should be more resilient to changes. WDYT?